src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -430,28 +449,71 @@ private static ParquetBufferValue getStructValue(
       SubColumnFinder subColumnFinder,
       ZoneId defaultTimezone,
       final long insertRowsCurrIndex,
-      boolean isDescendantsOfRepeatingGroup) {
+      boolean isDescendantsOfRepeatingGroup,
+      InsertValidationResponse.InsertError error) {
     Map<String, ?> structVal =
         DataValidationUtil.validateAndParseIcebergStruct(
             subColumnFinder.getDotPath(type.getId().toString()), value, insertRowsCurrIndex);
     Set<String> extraFields = new HashSet<>(structVal.keySet());
+    List<String> missingFields = new ArrayList<>();
```

Comment:
also need handling around non-nullable fields checking inside the struct?

Reply:
`IcebergParquetValueParser.parseColumnValueToParquet` will be called if and only if the value exist (could be null). The non-nullable error is handled there.